### DISCUSSION TITLE
Text decorations has no colorful background

### DISCUSSION THREAD

**Discussion by RuntimeErrorz** [JanDeDobbeleer/oh-my-posh#2532] [Troubleshoot] [answered]
I have managed to configure `oh-my-posh` successfully in one computer.  But after I executed the same configuration on another computer. The appreance looks weird. For example: the `powerlevel10k` theme should look like this.
![image](https://user-images.githubusercontent.com/71929239/179383931-b8210399-c1f3-4927-9281-928bd213c2cf.png)

But it actually looks like this.
![image](https://user-images.githubusercontent.com/71929239/179383939-9bd2bed4-37ea-43f7-9b67-256f78d05c3d.png)

The same problem also showed up using other themes. It seems that oh-my-posh somehow simplified all the characters for me.

I'm pretty sure that I did the same configuration. `settings.json` and `Microsoft.PowerShell_profile.ps1` are all the same.

I have also checked the `$PSStyle`, and both computers have the same `$PSStyle`，
![image](https://user-images.githubusercontent.com/71929239/179384022-acebd4a2-e759-4f0d-b044-e2c2ddbd068b.png)

I also reinstalled the Hack Nerd Font and FiraCode Nerd Font. I have searched the web but I can't find any answer.

The wrong rendering one is running Windows 10 20H2 19042.1110, 
And the correct one is running Windows 10 21H1 19043.1826.
The terminal app is both Windows Terminal Version 1.13.1143.

Any help would be appreciated.

**Comment by lewis-yeung** [comment]
@RuntimeErrorz What do you see when you run `oh-my-posh print primary`?

BTW, on the computer where the prompt is not properly rendered, does `$PSStyle.OutputRendering` output `Ansi` and `$Host.UI.SupportsVirtualTerminal` output `True`?

**Reply by RuntimeErrorz** [reply]
`oh-my-posh print primary` outputs the correct apperance like this.
![image](https://user-images.githubusercontent.com/71929239/179385268-9e85266c-e5df-4a7f-bff3-bcfe2738ab73.png)
And the `$PSStyle.OutputRendering` outputs `Host`, just like the successfully configured one.
But the `$Host.UI.SupportsVirtualTerminal` outputs `False` while the successfully configured one outputs `True`.

**Reply by JanDeDobbeleer** [reply]
@RuntimeErrorz can you set that to true and see if that changes something? You can clearly see that it can render it, something is only disabling it along the way.

**Reply by RuntimeErrorz** [reply]
@JanDeDobbeleer  It seems `'SupportsVirtualTerminal' is a ReadOnly property`.
![image](https://user-images.githubusercontent.com/71929239/179389606-88edeac7-70b0-4635-a619-a75bfa8664f5.png)

**Reply by lewis-yeung** [reply]
@RuntimeErrorz Are the two computers running the same OS and terminal app (e.g., Windows Terminal)?

**Reply by RuntimeErrorz** [reply]
The wrong rendering one is running Windows 10 20H2 19042.1110 , 
And the correct one is running Windows 10 21H1 19043.1826.
The terminal app is both Windows Terminal Version 1.13.1143.
Sorry to forget to mention it in the question descrption for I thought it's not relevant at first.

**Reply by lewis-yeung** [reply]
😕 That's so weird since the "virtual terminal processing" (see [MS docs](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences)) should have been enabled by default in Windows Terminal.

**Reply by JanDeDobbeleer** [reply]
@RuntimeErrorz can you ask at Windows Terminal? I know they have different renderers which could be the cause here.

**Reply by RuntimeErrorz** [reply]
OK, Thank you for your help.

**Reply by lewis-yeung** [reply]
@RuntimeErrorz 😔 No clue yet but definitely not an issue of OMP. I don't know if you are a fresh user to OMP or if it works in previous versions of Windows Terminal. It's been quite a while since I left Win10. I can only confirm that it works fine in Windows Terminal v1.13.11432.0 on Win11 21H2 (Build 22000.613).

**Comment by RuntimeErrorz** [comment, marked as answer]
It seems the system has a weird environment variable `$env:TERM` and its value is `dumb`, which causes `$Host.UI.SupportsVirtualTerminal = $false` as this [MSDoc](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.2) says. 
 I solved the problem by deleting this environment variable. But still I have no idea about how was this environment variable created.
